util/EIP-712/src/encode.rs

```diff
@@ -0,0 +1,414 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! EIP712 Encoder
```

Comment:
It would be nice to add api layout to the docs, something like this:
https://github.com/PaulRBerg/go-ethereum/blob/0b952c17c47d62381bdbb74ec1e1434a1906c7bf/signer/core/api_layout.md, although it's documented well in the EIP.